test/all/run_all.py

```diff
@@ -65,14 +65,19 @@ def __init__(self, device):
 
     def run(self):
         start_time = time.time()
-        if self.device.startswith("at"):
-            lbuild_command = ["-c", "avr.xml"]
-        elif self.device.startswith("stm32"):
-            lbuild_command = ["-c", "stm32.xml"]
-        lbuild_command.append("-D:target={}".format(self.device))
 
         with tempfile.TemporaryDirectory() as tempdir:
-            lbuild_command.extend(["-p", str(tempdir), "build", "--no-log"])
+            if self.device.startswith("at"):
+                lbuild_command = ["-c", "avr.xml"]
+                shutil.copyfile("avr.cpp", os.path.join(tempdir, "main.cpp"))
+            elif self.device.startswith("stm32"):
+                lbuild_command = ["-c", "stm32.xml"]
+                shutil.copyfile("stm32.cpp", os.path.join(tempdir, "main.cpp"))
```

Comment:
I feel like this should be done via lbuild too, maybe in a `:test:all` module, but then the unittest which do `:test:**` would also generate an additional `main.cpp` file. I'll fix it if it become annoying enough, this is fine.